src/interfaces.d.ts

```diff
@@ -56,6 +56,151 @@ export type MouseWheelEventHandler = (event?: MouseWheelEvent | WheelEvent) => E
+	 * This also includes attributes, styles, css classes. This callback is also invoked when node contains only text and that text changes.
+	 * {@link http://maquettejs.org/docs/animations.html|More about animations}.
+	 * @param element - Element that was modified in the DOM.
+	 * @param properties - The last properties object that was supplied to the [[h]] method
+	 * @param previousProperties - The previous properties object that was supplied to the [[h]] method
+	 */
+	updateAnimation?: (element: Element, properties?: VNodeProperties, previousProperties?: VNodeProperties) => void;
+	/**
+	 * Callback that is executed after this node is added to the DOM. Childnodes and properties have
+	 * already been applied.
+	 * @param element - The element that was added to the DOM.
+	 * @param projectionOptions - The projection options that were used see [[createProjector]].
+	 * @param vnodeSelector - The selector passed to the [[h]] function.
+	 * @param properties - The properties passed to the [[h]] function.
+	 * @param children - The children that were created.
+	 */
+	afterCreate?(element: Element, projectionOptions: ProjectionOptions, vnodeSelector: string, properties: VNodeProperties,
+	children: VNode[]): void;
+	/**
+	 * Callback that is executed every time this node may have been updated. Childnodes and properties
+	 * have already been updated.
+	 * @param element - The element that may have been updated in the DOM.
+	 * @param projectionOptions - The projection options that were used see [[createProjector]].
+	 * @param vnodeSelector - The selector passed to the [[h]] function.
+	 * @param properties - The properties passed to the [[h]] function.
+	 * @param children - The children for this node.
+	 */
+	afterUpdate?(element: Element, projectionOptions: ProjectionOptions, vnodeSelector: string, properties: VNodeProperties,
+	children: VNode[]): void;
+	/**
+	 * When specified, the event handlers will be invoked with 'this' pointing to the value.
+	 * This is useful when using the prototype/class based implementation of Components.
+	 *
+	 * When no [[key]] is present, this object is also used to uniquely identify a DOM node.
+	 */
+	readonly bind?: Object;
+	/**
+	 * Used to uniquely identify a DOM node among siblings.
+	 * A key is required when there are more children with the same selector and these children are added or removed dynamically.
+	 * NOTE: this does not have to be a string or number, a [[Component]] Object for instance is also possible.
+	 */
+	readonly key?: Object;
+	/**
+	 * An object literal like `{important:true}` which allows css classes, like `important` to be added and removed
+	 * dynamically.
+	 */
+	/*readonly classes?: { [index: string]: boolean | null | undefined };*/
+	readonly classes?: {
+		[index: string]: boolean | null | undefined;
+	} | ClassesFunction;
+	/**
+	 * An object literal like `{height:'100px'}` which allows styles to be changed dynamically. All values must be strings.
```

Comment:
May want to note that this is not encourages in most cases? Or is is fine to do this?